### PR TITLE
changes to no validate comment

### LINKED ISSUE #34809: Need Regex to escape comments while validating the solution
**Describe your problem and how to reproduce it:**
The solution validates the comments which checking for "for" keyword. I think it should escape the comments.
I have added a small screenshot with solution. not sure if its 100% correct. but hopefully something to start with.

**Add a Link to the page with the problem:**
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array/

**Tell us about your browser and operating system:**
* Browser Name: Chrome
* Browser Version: Version 71.0.3578.98 (Official Build) (64-bit)
* Operating System: Windows 10 pro

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
screenshot 93
screenshot 92

### PR DESCRIPTION
**Pull request by Vikas-KM**
Changed regex to not validate comment.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #34809

### CODE REVIEW

**Comment by RandellDawson** [comment]
What if someone uses `/*   */` like:

```
/*
for (var i = 0; i <watchList.length; i++) {
*/
```
or

```
/* for (var i = 0; i <watchList.length; i++) {
*/
```
or
```
/*  for (var i = 0; i <watchList.length; i++) { */
```

**Comment by RandellDawson** [comment]
In the future, it is best to discuss all the edge cases on the original issue and make sure you have buy-in from others before creating a PR.  It may be deemed not something that is needed or the discussion can yield other things to think about (such as what I present above).

**Comment by Vikas-KM** [comment]
> What if someone uses `/* */` like:
> 
> ```
> /*
> for (var i = 0; i <watchList.length; i++) {
> */
> ```
> or
> 
> ```
> /* for (var i = 0; i <watchList.length; i++) {
> */
> ```
> or
> 
> ```
> /*  for (var i = 0; i <watchList.length; i++) { */
> ```
> You are trying to hack the test because you want to put a solution in comments? There may be other edge cases here. While I understand your intent, it seems like an necessary modification to the tests for a very unique case which is specific just to your code.



> What if someone uses `/* */` like:
> 
> ```
> /*
> for (var i = 0; i <watchList.length; i++) {
> */
> ```
> or
> 
> ```
> /* for (var i = 0; i <watchList.length; i++) {
> */
> ```
> or
> 
> ```
> /*  for (var i = 0; i <watchList.length; i++) { */
> ```
> You are trying to hack the test because you want to put a solution in comments? There may be other edge cases here. While I understand your intent, it seems like an necessary modification to the tests for a very unique case which is specific just to your code.

Sure that regex was not made keeping in block comments, which i am working, in the original issue they said to make a PR and lets discuss further hence i created PR. 
Sorry,  I will try to make sure i get buy-in and validate edge cases before making PR.

**Comment by RandellDawson** [comment]
@Vikas-KM The advise I gave regarding discussing the issue first, is just to prevent you from possibly spending/wasting a lot of time for something that others may feel is a non-issue or not worth pursuing.  I am curious as to what others think about writing tests for such an edge case.

**Comment by RandellDawson** [comment]
@Vikas-KM After thinking about this a bit more and looking at some other challenges where comments where removed and not validated in the solutions, I suggest you create a "helper" function which can be placed in what is referred to as the "After Test" section of the challenge seed.  To do this, you could use something like below to accomplish this.  There are some edge cases which it might miss, but should at least take care of most single and multi-line comments in the JavaScript code which might be tried in this particular challenge.

Within the `section` with id="challengeSeed" (below existing , `js-seed` div, you would have the following.
````
### After Test
<div id='js-teardown'>

```js
const removeJSComments = str => str.replace(/\/\*[\s\S]*?\*\/|\/\/.*$/gm, '');
```

</div>
````

Then, you can simplify the assert to the following for the second test:
```yml
  - text: Your code should not use a <code>for</code> loop.
    testString: assert(!removeJSComments(code).match(/for\s*?\(.+?\)/), 'Your code should not use a <code>for</code> loop.');
```

Please locally test various comments with `for` loops to make sure you are satisfied it does what you were hoping to resolve in issue #34809.  If so, please make the changes and commit them to this PR.

**Comment by RandellDawson** [comment]
So I have made my suggested changes on behalf of the user.

@Manish-Giri, @thecodingaviator Can you review?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md [hunk 1]
```diff
@@ -24,13 +24,13 @@ The <code>watchList</code> array holds objects with information on several movie
 ```yml
 tests:
   - text: The <code>watchList</code> variable should not change.
-    testString: assert(watchList[0].Title === "Inception" && watchList[4].Director == "James Cameron", 'The <code>watchList</code> variable should not change.');
+    testString: assert(watchList[0].Title === "Inception" && watchList[4].Director == "James Cameron");
   - text: Your code should not use a <code>for</code> loop.
-    testString: assert(!code.match(/for\s*?\(.+?\)/g), 'Your code should not use a <code>for</code> loop.');
+    testString: assert(!removeJSComments(code).match(/for\s*?\(.+?\)/));
```
Comment (thecodingaviator) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md]:
```suggestion
    testString: assert(!removeJSComments(code).match(/for\s*?\(.*?\)/));
```
I guess this is better than what we have, this will match empty `for`s too. Since the test is `Your code should not use a <code>for</code> loop` even an empty for is technically a for

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md]: @thecodingaviator Fair enough.

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md]: @thecodingaviator Made the change.

**Review by Manish-Giri — APPROVED** [review]
@Vikas-KM @RandellDawson LGTM. Thank You for your contribution.
